src/main/scala/beam/agentsim/infrastructure/ChargingNetworkManager.scala

```diff
@@ -0,0 +1,249 @@
+package beam.agentsim.infrastructure
+
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import beam.agentsim.agents.modalbehaviors.DrivesVehicle._
+import beam.agentsim.agents.vehicles.BeamVehicle
+import beam.agentsim.infrastructure.power.{PowerController, SitePowerManager}
+import beam.agentsim.scheduler.BeamAgentScheduler.{CompletionNotice, ScheduleTrigger}
+import beam.agentsim.scheduler.Trigger
+import beam.agentsim.scheduler.Trigger.TriggerWithId
+import beam.sim.config.BeamConfig
+import beam.sim.{BeamScenario, BeamServices}
+import beam.utils.DateUtils
+import org.matsim.api.core.v01.Id
+
+import scala.collection.concurrent.TrieMap
+
+class ChargingNetworkManager(
+  beamServices: BeamServices,
+  beamScenario: BeamScenario,
+  scheduler: ActorRef
+) extends Actor
+    with ActorLogging {
+  import ChargingNetworkManager._
+
+  private val beamConfig: BeamConfig = beamScenario.beamConfig
+  private val vehiclesToCharge: TrieMap[Id[BeamVehicle], ChargingVehicle] = new TrieMap()
+  private def vehicles: Map[Id[BeamVehicle], BeamVehicle] = vehiclesToCharge.mapValues(_.vehicle).toMap
+
+  private val sitePowerManager = new SitePowerManager()
+  private val powerController = new PowerController(beamServices, beamConfig)
+  private val endOfSimulationTime: Int = DateUtils.getEndOfTime(beamConfig)
+
+  private val gridConnectionEnabled = beamConfig.beam.agentsim.chargingNetworkManager.gridConnectionEnabled
+  log.info("ChargingNetworkManager should be connected to grid: {}", gridConnectionEnabled)
```

Comment:
A bit vice versa:

On the top level we have a configuration whether `gridConnectionEnabled` should be enabled or not.

This means if it's `true` then we will try to connect to federate grid if it's possible. But this is a responsibility of `powerController` to try to establish connection with Helics and connect to the grid. That's why we are calling their method `isConnectedToGrid` ( - probably better solution is to rename this method on somethign `initConnection`)

But if `gridConnectionEnabled == false` then we even don't want to force `powerController` to establish connection with Helics and that's why we call its method `powerController.defaultPowerPhysicalBounds()` which is safe to call. I will add a comment to code about that

Reply:
I agree the name change helps, but I wonder if the config should be flipped - `enableGridConnection` or `allowGridConnection` or `attemptGridConnection` - or something of that sort? It would probably fully close the loop more concretely.